FragmentAggregator improved ers

MRiganSUSX opened the pull request:
Improving warning message when sender->send fails (request by @mroda88).

- passes unit test
- passes integ test bundle
- tested locally by manually triggering the catch block:
![image](https://github.com/user-attachments/assets/d3cf77ae-71cb-419d-8e2a-b256eea7867d)

MRiganSUSX committed:
adding more verbose error when sender send fails

MRiganSUSX committed:
clang

MRiganSUSX commented:
@mroda88 
Few questions:
- do we want this to remain a `warning`
- are we sure the `sender->send` throws? (otherwise what is the point of this catch)
- the `data_request.data_destination` seems to be an empty string

mroda88 commented on src/dfmodules/CommonIssues.hpp:
```diff
@@ -87,6 +87,22 @@ ERS_DECLARE_ISSUE(dfmodules,         ///< Namespace
+                    << seqno << ", destination: " << datadest << ", requested component: " << sid << ".",
+                  ((daqdataformats::run_number_t)runno)      ///< Message parameters
+                  ((daqdataformats::trigger_number_t)trigno) ///< Message parameters
+                  ((daqdataformats::timestamp_t)triggerts)   ///< Message parameters
```
I think we can slim down and remove the trigger time. It's not necessary to uniquely identify the data request.

mroda88 commented:
For your questions. 
 - no, it must be turned into an error because it's data loss (eventually)
 -  We are sure, it can throw
 - I think we can simplify and get rid of the destination, we already have the SourceId, that's enough.

MRiganSUSX committed:
simplified msg, changed to error

MRiganSUSX commented:
- changed to error
- removed parts you mentioned

test:
![image](https://github.com/user-attachments/assets/176b0e09-9c32-411d-8ec6-d600b8693f5c)

otherwise passed unit tests and integtests bundle.

mroda88 commented:
great. Now if you could also bump the patch version, we can merge that in too. And we will just need to tag.

MRiganSUSX committed:
bumping the patch version to v4.1.2

mroda88 reviewed (approved):
Great, thanks
